source/appModules/code.py

```diff
@@ -21,11 +24,92 @@ class VSCodeDocument(Document):
 	_get_treeInterceptorClass = NVDAObject._get_treeInterceptorClass
 
 
+DIGIT_EXPR = re.compile(r"\d+")
+
+
 class AppModule(appModuleHandler.AppModule):
+	def __init__(self, *args, **kwargs):
+		super().__init__(*args, **kwargs)
+		self._status = None
+
+	@staticmethod
+	def _search_for_statusbar(root: NVDAObject) -> NVDAObject | None:
+		seen = set()
+		t = deque((root,))
+		while t:
+			obj = t.popleft()
+			if obj in seen:
+				continue
+			seen.add(obj)
+			try:
+				if obj.role == controlTypes.Role.STATUSBAR:
+					return obj
+				# IA2 ID often contains "statusbar"
+				ia2id = obj.IA2Attributes.get("id")
+				if ia2id and "statusbar" in ia2id.casefold():
+					return obj
+				children = obj.children
+			except Exception:
+				children = ()
```

Comment:
```suggestion
			if obj.role == controlTypes.Role.STATUSBAR:
				return obj
			# IA2 ID often contains "statusbar"
			ia2id = obj.IA2Attributes.get("id")
			if ia2id and "statusbar" in ia2id.casefold():
				return obj
			try:
				children = obj.children
			except Exception:
				log.exception()
				children = ()
```

Reply:
Can you please restore this change?

Reply:
Suppose the search somehow lands on a UIA object:

* With bc76c79: The exception `AttributeError: 'UIA' object has no attribute 'IA2Attributes'` is raised and the entire search fails.
* With e1770dcd90: The exception is logged at level error, and search continues omitting this level (which might still be successful).

Reply:
Then this should be

```suggestion
			if obj.role == controlTypes.Role.STATUSBAR:
				return obj
			# IA2 ID often contains "statusbar"
			try:
				ia2id = obj.IA2Attributes.get("id")
			except AttributeError:
				ia2id = None
			if ia2id and "statusbar" in ia2id.casefold():
				return obj
			try:
				children = obj.children
			except Exception:
				log.exception()
				children = ()
```